### PR TITLE
Part - persistant stock history

### PR DESCRIPTION
**Pull request by matmair**
for #1484

### CODE REVIEW

**Comment by SchrodingersGat** [comment]
@matmair can you provide some more context details here, what you are trying to achieve?

**Comment by matmair** [comment]
@SchrodingersGat Sure
TLDR: This PR enables a graph of historical stock levels even when stock get deleted as you suggested in #1484.


This PR implements a new model `PartQuantityHistory` to which an entry is added each time a StockItemTracking entry is added (via signal). 

In this new model there are fields for the data tracked in StockItemTracking and total stock at time of creation (after the action triggering the tracking was processed). This information could be used for graphs but also be enhanced with more tracked properties to make more historical insights possible.

As the tracking models share a lot of similarities I made as many parts of them abstract/reusable as possible:
This included:
Python:  `ItemTracking` an abstract model as basis for tracking models, `TrackingMixin` a mixin for API views with tracking models, `TrackingSerializer` a serializer base class for tracking models.
JS: `loadTrackingTable` shared table-loading function for both StockIteTracking and PartQuantityHistory.

From a UI perspective there are new API and View-Url-Endpoints, a navbar entry and a view for viewing the tracking entries on the part-detail views - based on how it is done in StockItems.

**Comment by matmair** [comment]
@SchrodingersGat do we even need this feature anymore? This was meant to fix #1484

**Comment by SchrodingersGat** [comment]
@matmair sorry I haven't been giving this one much attention. I'm not sure that this feature fits in *right now* - I've been meaning to spend some time looking at refactoring the existing stock tracking history, which means I was overlooking this PR as I thought it would change it anyway.

I think at the moment you'd be better served focusing your attention on other components! If you're happy to close this issue for now, we can maybe come back to it at a later date. But currently I think just keeping it up to date with master is a waste of time :)

**Comment by matmair** [comment]
@SchrodingersGat glad we feel the same way. I am also not really sure anybody needs this function. I certainly don't - would be nice but not necessary.
